static/js/directives/chat.js

```diff
@@ -336,6 +340,19 @@ define(['jquery', 'underscore', 'text!partials/chat.html', 'text!partials/chatro
 						subscope.doClear = function() {
 							subscope.$broadcast("clear");
 						};
+						subscope.addContact = function() {
+							subscope.$emit("requestcontact", subscope.id, {
+								restore: true
+							});
+						};
+						subscope.removeContact = function() {
+							contacts.remove(buddy.contact.Userid);
+						};
+						subscope.updateContactStatus = function(event, data) {
+							subscope.isContact = event.type === "contactadded";
+						};
+						contacts.e.on("contactadded", subscope.updateContactStatus);
+						contacts.e.on("contactremoved", subscope.updateContactStatus);
```

Comment:
Am filtering now by `userid` to correctly handle this. Not sure what else I would use to handle this, other then the contacts api.